apps/server/src/modules/authentication/strategy/ldap.strategy.ts

```diff
@@ -81,13 +81,31 @@ export class LdapStrategy extends PassportStrategy(Strategy, 'ldap') {
 		}
 	}
 
-	private async loadAccount(username: string, system: System): Promise<AccountDto> {
-		const ldapConfig: LdapConfig = this.checkValue(system.ldapConfig);
-		const rootPath: string = this.checkValue(ldapConfig.rootPath);
-
-		const userNameWithSchool = `${rootPath}/${username}`;
-
-		const account = await this.authenticationService.loadAccount(userNameWithSchool.toLowerCase(), system.id);
+	private async loadAccount(username: string, systemId: string, school: SchoolDO): Promise<AccountDto> {
+		const externalSchoolId = this.checkValue(school.externalId);
+
+		let account: AccountDto;
+
+		// TODO having to check for two values in order to find an account is not optimal and should be changed.
+		// The way the name field of Accounts is used for LDAP should be reconsidered, since
+		// mixing the login name with a technical id from a foreign system is not a good pattern.
+		// Binding the login name to an identifier from a foreign system or an identifier of a school can lead to
+		// accounts not being found when the identifier changes.
+		if (school.previousExternalId) {
+			try {
+				account = await this.authenticationService.loadAccount(
+					`${externalSchoolId}/${username}`.toLowerCase(),
+					systemId
+				);
+			} catch (e: unknown) {
+				account = await this.authenticationService.loadAccount(
+					`${school.previousExternalId}/${username}`.toLowerCase(),
+					systemId
+				);
+			}
+		} else {
+			account = await this.authenticationService.loadAccount(`${externalSchoolId}/${username}`.toLowerCase(), systemId);
+		}
```

Comment:
we have the same statement for account load on both the if and the else :)


also the catch unknown error, at least we should try to log something